Apps/Sandcastle/CesiumSandcastle.js

```diff
@@ -810,6 +811,14 @@ require({
         }
     }
 
+    registry.byId('buttonNew').on('click', function() {
+        loadFromGallery(newDemo);
+        var demoSrc = newDemo.name + '.html';
+        if (demoSrc !== window.location.search.substring(1)) {
```

Comment:
That's what I assumed, but it doesn't actually work in this case.  Just open up Sandcastle and hit the new button a few times and you'll see it's added to the history each time.  @emackey, any thoughts on a better way to do this detection?